### PR TITLE
ceph.spec.in: add support for openEuler OS

### PR DESCRIPTION
**Pull request by liuqinfei**
Currently, as the openEuler OS has been used by more and more people. Users expect Ceph to support the compilation on the openEuler.

Signed-off-by: liuqinfei [lucas.liuqinfei@huawei.com](mailto:lucas.liuqinfei@huawei.com)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by liuqinfei** [comment]
jenkins retest this please

**Review by idryomov — APPROVED** [review]

**Comment by rzarzynski** [comment]
Might be worth renaming the commit to `ceph.spec.in: add support for openEuler OS`.

Also, putting info on testing the change (full testing procedure would be supercool!) can help.

Generally LGTM!

**Comment by ljflores** [comment]
jenkins test api

**Comment by liuqinfei** [comment]
> Might be worth renaming the commit to `ceph.spec.in: add support for openEuler OS`.
> 
> Also, putting info on testing the change (full testing procedure would be supercool!) can help.
> 
> Generally LGTM!

Completed the updating of the commit information. Please review again. Thanks.

**Comment by idryomov** [comment]
Can you provide a link to a Storage SIG build dashboard (something listing builds and their statuses, similar to e.g. https://shaman.ceph.com/builds/ceph/), if such a thing exists?

**Comment by liuqinfei** [comment]
> Can you provide a link to a Storage SIG build dashboard (something listing builds and their statuses, similar to e.g. https://shaman.ceph.com/builds/ceph/), if such a thing exists?

We have tested with the qa suits. Here is the verification report summary. 
https://gitee.com/src-openeuler/ceph/wikis/3%20-%20%E5%88%86%E5%B8%83%E5%BC%8F%E5%AD%98%E5%82%A8%E7%BB%84%E4%BB%B6%E4%BD%BF%E8%83%BD/ceph%E8%B4%A8%E9%87%8F%E7%9C%8B%E6%8A%A4qa
The qa platform currently may have some troubles, i will check later. But, in fact, Ceph 12.2.8 was released and provided in openEuler 20.03 and Ceph 16.2.7 was released and provided in openEuler 22.03.

**Comment by liuqinfei** [comment]
And we have the ceph daily build on openEuler.
The OS version is openEuler22.03 LTS SP1.
[https://github.com/openeuler-mirror/ceph-daily-build](https://github.com/openeuler-mirror/ceph-daily-build)

**Review by ljflores — COMMENTED** [review]
Hey @liuqinfei, please drop the extra merge commit. Also, as per the discussion at CDM, any documentation changes can also be added to this PR.

**Comment by liuqinfei** [comment]
> Hey @liuqinfei, please drop the extra merge commit. Also, as per the discussion at CDM, any documentation changes can also be added to this PR.
> 
> Thanks for your contribution!

I have updated the PR and included the information about the ceph versions released in openEuler. Please have a look. Thanks.

**Review by Svelar — APPROVED** [review]

File: doc/install/get-packages.rst [hunk 1]
```diff
@@ -251,6 +251,17 @@ openSUSE Tumbleweed
...
+^^^^^^^^^
+
+There are two major versions supported in normal openEuler repositories. They are ceph 12.2.8 in openEuler-20.03-LTS series and ceph 16.2.7 in openEuler-22.03-LTS series. There’s no need to add another package repository manually.
+You can install ceph just by executing the following:
+
+.. prompt:: bash $
+
+    sudo yum -y install ceph
+
+Also you can download packages nanually from https://repo.openeuler.org/openEuler-{release}/everything/{arch}/Packages/.
```
Comment (idryomov) [inline doc/install/get-packages.rst:264]:
```suggestion
Also you can download packages manually from https://repo.openeuler.org/openEuler-{release}/everything/{arch}/Packages/.
```

Comment (liuqinfei) [reply doc/install/get-packages.rst]: Fix that. Thanks.

**Comment by wangxiyuan** [comment]
jenkins test api

**Comment by wangxiyuan** [comment]
@ljflores Hi, this PR need your review again. Thanks!

**Comment by kevinzs2048** [comment]
Hi @idryomov @ljflores, appreciated for your help on this PR. Now @liuqinfei has update the patchset, and pass all the CI jobs. Please help to review again when you convenient.

**Comment by idryomov** [comment]
@kevinzs2048 My approval stands.

@yuriw This PR has had `needs-qa` label for over three months now.  Could you please pick it up into the next RADOS integration branch?  No specific testing is required, just a sanity check around RPM packages still building and installing.

**Review by ljflores — APPROVED** [review]
LGTM!

**Comment by ljflores** [comment]
Hey @liuqinfei this is now going in for testing. We will just verify the build and let you know the results!

**Comment by ljflores** [comment]
@liuqinfei results were good, and your contribution has been merged. Thanks so much! :)

**Pull request merged**